tests/targets/test_android.py

```diff
@@ -52,11 +46,57 @@ def patch_platform(platform):
     return mock.patch("buildozer.targets.android.platform", platform)
 
 
-class TestTargetAndroid:
+def init_target(temp_dir, options=None):
+    buildozer = init_buildozer(temp_dir, "android", options)
+    return TargetAndroid(buildozer)
 
-    @staticmethod
-    def default_specfile_path():
-        return os.path.join(os.path.dirname(buildozer_module.__file__), "default.spec")
+
+def call_build_package(target_android):
+    """
+    Call the build_package() method of the tested TargetAndroid instance,
+    patching the functions that would otherwise produce side-effects.
+
+    Return the mocked execute_build_package() method of the TargetAndroid
+    instance so that tests can easily check which command-line arguments
+    would be passed on to python-for-android's toolchain.
+    """
+    buildozer = target_android.buildozer
+    expected_dist_dir = (
+        '{buildozer_dir}/android/platform/build-armeabi-v7a/dists/myapp__armeabi-v7a'.format(
+        buildozer_dir=buildozer.buildozer_dir)
+    )
+
+    with patch_target_android(
+        '_update_libraries_references'
+    ) as m_update_libraries_references, patch_target_android(
+        '_generate_whitelist'
+    ) as m_generate_whitelist, mock.patch(
+        'buildozer.targets.android.TargetAndroid.execute_build_package'
+    ) as m_execute_build_package, mock.patch(
+        'buildozer.targets.android.copyfile'
+    ) as m_copyfile, mock.patch(
+        'buildozer.targets.android.os.listdir'
+    ) as m_listdir:
+        m_listdir.return_value = ['30.0.0-rc2']
+        target_android.build_package()
```

Comment:
I really dislike this with statement formatting :<

Reply:
sure, let me know the way you like it so I can change it to it

Reply:
I've updated it to the version with the backslash https://github.com/psf/black/blob/7403d95/docs/the_black_code_style.md#how-black-wraps-lines let me know what you think